packages/stack/proxy/src/proxy.js

```diff
@@ -44,12 +39,12 @@ export class Proxy {
   }
 
   async _createWebSocketProvider(endpoint) {
-    // if we are using a VM (ie for tests), then try to get the VM provider
-    if (this.isVm) {
-      return this.events.request2("blockchain:client:vmProvider");
+    // if we are using a VM (ie for tests), then get the VM provider
+    if (this.vm) {
+      return this.vm;
     }
```

Comment:
I don't think we should include this, and instead modify `blockchain:client:provider` (see overall review comment).